src/psyclone/gocean1p0.py

```diff
@@ -880,6 +881,222 @@ def backward_dependence(self):
+                for idx0 in range(3):
+                    for idx1 in range(3):
+                        self._stencil[idx0][idx1] = int(args[idx1][idx0])
+            else:
+                # stencil info is of the form 'name' so should be one
+                # of our valid names
+                if name not in VALID_STENCIL_NAMES:
+                    raise ParseError(
+                        "Meta-data error in kernel '{0}': 3rd descriptor "
+                        "(stencil) of field argument is '{1}' but must be one "
+                        "of {2} or stencil(...)".format(kernel_name, name,
+                                                        VALID_STENCIL_NAMES))
+                self._name = name
+                # We currently only support one valid name ('pointwise')
+                # which indicates that there is no stencil
+                self._has_stencil = False
+        else:
+            # the stencil information is not in the expected format
+            raise ParseError(
+                "Meta-data error in kernel '{0}': 3rd descriptor (stencil) of "
+                "field argument is '{1}' but expected either a name or the "
+                "format 'stencil(...)'".format(kernel_name, str(stencil_info)))
+
+    def _check_init(self):
+        '''Internal method which checks that the stencil information has been
+        loaded.
+
+        :raises GenerationError: if the GOStencil object has not been
+        initialised i.e. the load() method has not been called
+
+        '''
+        if not self._initialised:
+            raise GenerationError(
+                "Error in class GOStencil the object has not yet been "
+                "initialised. Please ensure the load() method is called.")
+
+    @property
+    def has_stencil(self):
+        '''Specifies whether this argument has stencil information or not. The
+        only case when this is True is when the stencil information
+        specifies 'pointwise' as this indicates that there is no
+        stencil access.
+
+        :return Bool: True if this argument has stencil information
+        and False if not.
+
+        '''
+        self._check_init()
+        return self._has_stencil
+
+    @property
+    def name(self):
+        '''Provides the stencil name if one is provided
+
+        :return string: the name of the type of stencil if this is
+        provided and 'None' if not.
+
+        '''
+        self._check_init()
+        return self._name
+
+    def depth(self, index0, index1):
+        '''Provides the depth of the stencil in the 8 possible stencil
+        directions in a 2d regular grid (see the description in the
+        load class for more information). Values must be between -1
+        and 1 as they are considered to be relative to the centre of
+        the stencil For example:
+
+        stencil(234,
+                915,
+                876)
+
+        returns
+
+        depth(-1,0) = 9
+        depth(1,1) = 4
+
```

Comment:
Please document the arguments to this method.

Reply:
Oh yes I missed these